cpp/arcticdb/storage/failure_simulation.hpp

```diff
@@ -77,29 +77,51 @@ inline std::ostream& operator<<(std::ostream& out, const FailureAction& action)
     return out;
 }
 
-namespace action_factories { // To allow `using namespace`
+namespace action_factories {
+// To allow `using namespace`
 static inline const FailureAction no_op("no_op", [](FailureType){});
 
+static FailureAction::FunctionWrapper maybe_execute(double probability, FailureAction::FunctionWrapper&& func) {
+    util::check_arg(probability >= 0 && probability <= 1.0, "Bad probability: {}", probability);
+
+    return [probability, f = std::move(func)](FailureType type) mutable {
+        ARCTICDB_DEBUG(log::lock(), "Probability to use StorageFailureSimulator, {}", probability);
+        if (probability == 0) {
+            return;
+        }
+
+        if (probability == 1.0) {
+            std::invoke(std::move(f), type);
+            return;
+        }
+
+        thread_local std::uniform_int_distribution<size_t> dist(0.0, 1.0);
+        thread_local std::mt19937 gen(std::random_device{}());
```

Comment:
Changed to this random generator as old one gave me only probabilities up to 0.5 so might be something with it. We can address this as part of the failure simulation task.